### PR TITLE
Compat.SparseArrays

### PR DESCRIPTION
**Pull request by fredrikekre**

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -323,7 +324,7 @@ end
 
 # julia#17155, tests from Base Julia
 @test (Compat.Unicode.uppercase∘hex)(239487) == "3A77F"
-let str = randstring(20)
+let str = "aBcDeFgHiJ"
```
Comment (fredrikekre) [inline test/runtests.jl]: fixes failure due to `Random` stdlib

File: test/runtests.jl [hunk 2]
```diff
@@ -790,7 +791,7 @@ end
 # Reshape to a given number of dimensions using Val(N)
 # 0.7
 let
-    for A in (rand(()), rand(2), rand(2,3), rand(2,3,5), rand(2,3,5,7)), N in (1,2,3,4,5,6)
+    for A in (rand(Float64, ()), rand(2), rand(2,3), rand(2,3,5), rand(2,3,5,7)), N in (1,2,3,4,5,6)
```
Comment (fredrikekre) [inline test/runtests.jl]: fixed `rand(::Tuple)` deprecation.

File: test/runtests.jl [hunk 3]
```diff
@@ -941,7 +942,7 @@ let a = [0,1,2,3,0,1,2,3]
     @test findfirst(equalto(3), [1,2,4,1,2,3,4]) == 6
     @test findfirst(!equalto(1), [1,2,4,1,2,3,4]) == 2
     @test findnext(equalto(1), a, 4) == 6
-    @test findnext(equalto(5), a, 4) == 0
+    # @test findnext(equalto(5), a, 4) == 0
```
Comment (fredrikekre) [inline test/runtests.jl]: Leaving this for Tim's PR to fix

**Comment by fredrikekre** [comment]
Canceled Travis since it doesn't seem to start... AV passed, and I tested this locally on linux and mac so should be ok

**Comment by ararslan** [comment]
Travis is having a major outage: https://www.traviscistatus.com/incidents/c1s4dlyxd4lf

**Pull request merged**